keychain/src/keychain.rs

```diff
@@ -33,6 +33,45 @@ pub struct ExtKeychain {
 	hasher: BIP32GrinHasher,
 }
 
+impl ExtKeychain {
+	fn derive_key_internal(&self, amount: u64, id: &Identifier, switch: bool) -> Result<SecretKey, Error> {
+		let mut h = self.hasher.clone();
+		let p = id.to_path();
+		let mut ext_key = self.master;
+		for i in 0..p.depth {
+			ext_key = ext_key.ckd_priv(&self.secp, &mut h, p.path[i as usize])?;
+		}
+
+		match switch {
```

Comment:
I realize you are just moving this code but why would we not always use switch commitments?  If wallets are creating outputs without using switch commits they risk the possibility of losing funds if Grin is ever forced to require revealing the pre-switch commit thanks to QC or smth.